plugins/askrene/mcf.c

```diff
@@ -1457,6 +1458,10 @@ linear_routes(const tal_t *ctx, struct route_query *rq,
 			/* Leave exactly 1 slot for the next round of
 			 * computations. */
 			excess_parts = 1;
+			/* If we only *had* one slot, remove the channel which refinement said limited
+			 * the flow, so we don't use it again and infinite loop. */
+			if (parts_slots == 1)
+				bitmap_set_bit(rq->disabled_chans, bottleneck_idx);
```

Comment:
Looks good. But on the extra careful side, are we sure there is always going to be a blottlenect_idx? ie. that the previous function will always write on that un-initialized variable?